### PR TITLE
Config file merging & test & default AWS region

### PR DESCRIPTION
**Pull request by karimodm**

### CODE REVIEW

File: test/config/config-test.js [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+  it('overrides properties on merge', () => {
+    fs.existsSync.returns(false)
+    readFile.onSecondCall().returns( { aws: { region: 'us-west-2' } } )
+
+    const config = configFactory(fs, readFile, 'non-existent-env')
+
+    expect(config.aws.region).to.equal('us-west-2')
+    sinon.assert.calledTwice(readFile)
+  })
+
```
Comment (alongalky) [inline test/config/config-test.js]: extra whitespace :) (MS style code review)

File: config/config.js [hunk 2]
```diff
@@ -1,16 +1,15 @@
-const fs = require('fs')
 const path = require('path')
+const objectMege = require('object-merge')
```
Comment (alongalky) [inline config/config.js]: I recommend using Object.assign instead of 'object-merge'. It's built in and does the same

File: package.json [hunk 3]
```diff
@@ -23,6 +23,7 @@
     "express": "^4.15.2",
     "express-validator": "^3.1.2",
     "mysql2": "^1.2.0",
+    "object-merge": "^2.5.1",
```
Comment (alongalky) [inline package.json]: Your call, but if you accept my advice about Object.assign then remove this dependency

File: test/config/config-test.js (hunk 1 above)
Comment (karimodm) [inline test/config/config-test.js]: won't fix

**Pull request merged**